include/deal.II/numerics/coarsening_strategies.h

```diff
@@ -0,0 +1,94 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_coarsening_strategies_h
+#define dealii_coarsening_strategies_h
+
+
+#include <deal.II/base/config.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * When data is transferred during coarsening, it is not trivial to decide
+ * how to handle data of child cells which will be coarsened. Or in other
+ * words, which data should be stored in the corresponding parent cell.
+ *
+ * In this namespace, we offer a few strategies that cope with this
+ * problem. Such strategies can be passed to the CellDataTransfer and
+ * parallel::distributed::CellDataTransfer constructors.
+ *
+ * @author Marc Fehling, 2019
+ */
+template <typename value_type>
```

Comment:
Do we have some template type check we could use here to ensure that `median` would work correctly?

Reply:
Were you referring to the `mean` function? I guess the code wouldn't compile if `value_type` has no zero value and no `operator+=`.

Otherwise I don't know what you mean by `median`. Would you elaborate on that?

Reply:
Are you suggesting a new `median` function that returns the median of all child values? (The number of children is always even)

Reply:
Sorry, I meant `mean`.

Reply:
We may use the  concepts feature that has been introduced in [C++20](https://en.cppreference.com/w/cpp/language/constraints) or [Boost 1.4](https://www.boost.org/doc/libs/1_70_0/libs/concept_check/concept_check.htm) for this.

But I think using [type traits](https://en.cppreference.com/w/cpp/types) to check for arithmetic types should be sufficient for now until someone demands a more versatile implementation. But I don't know how to sum/mean `bool` variables - we may just exclude them.